Refreshed & Enhanced UI

sunil32150 opened the pull request:
#16 # What is this PR for?
In this version you will find a new fresh ui/look and feel, enhanced side bar, multiple theme, new login page, Dashboard page enhanced, Widget function enhanced, every page is now data-grid functioned , add edit delete is much easier and have enhanced look. Role and user module is enhanced and shown in different pages.


### What type of PR is it?
Improvement  & eature

### What is the Github issue?
* Open an issue on Github https://github.com/yzhang921/CBoard/issues
* Put link here, and add [- #ISSUE_NUMBER] in PR title
eg. [ES] Auto adjust time interval bucket when override of date column is null #114

### How should this be tested?
Just Build and run

### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/40990590/61131083-e183a080-a4d5-11e9-9e15-3a5d5ddcf170.png)
![image](https://user-images.githubusercontent.com/40990590/61131157-0d9f2180-a4d6-11e9-90f3-d18be3cd5b38.png)
![image](https://user-images.githubusercontent.com/40990590/61131209-36bfb200-a4d6-11e9-8b9b-691daf2b8ed0.png)

![image](https://user-images.githubusercontent.com/40990590/61131220-40e1b080-a4d6-11e9-9536-2028896b5247.png)

And other places have also been changed to list view.

sunil32150 committed:
Refreshed & Enhanced UI

In this version you will find a new fresh ui/look and feel, enhanced side bar, multiple theme, new login page, Dashboard page enhanced, Widget function enhanced, every page is now data-grid functioned , add edit delete is much easier and have enhanced look. Role and user module is enhanced and shown in different pages.

sunil32150 committed:
Revert back to original

sunil32150 committed:
Update pom.xml

Changes in dependecies

yzhang921 commented:
Great job, we need to take some time to do detail code review. But there are some obvious problems I can point out.
1 it would be better to extract the duplicated code of pagination.
2 Please replace all the tab with 4 white spaces
3 Looks like you have missing files to commit
```
http://localhost:8025/imgs/syncGrid.png 404
http://localhost:8025/imgs/syncGrid-active.png 404
http://localhost:8025/org/cboard/view/config/chart/grid.html 404
http://localhost:8025/org/cboard/view/config/chart/options/grid.html 404
```
4 Change ES6 style arrow function to ES5 normal function at the following files, that may cause some low version chrome can't compatible:
 - userAdminCtrl.js
 - dashboardViewCtrl.js
 - widgetCtrl.js
5 I don't think it's a good idea to remove the jstree resource style of widget/dataset
6 and also user, role and resource page split is not convenient to manged in separate page.

yzhang921 commented:
I will help to forward this pr to more user group to involve in voting, thanks!

sunil32150 committed:
Issue from cboard pull request resolved (yzhang921 commented )

1 it would be better to extract the duplicated code of pagination.
2 Please replace all the tab with 4 white spaces
3 Looks like you have missing files to commit

http://localhost:8025/imgs/syncGrid.png 404
http://localhost:8025/imgs/syncGrid-active.png 404
http://localhost:8025/org/cboard/view/config/chart/grid.html 404
http://localhost:8025/org/cboard/view/config/chart/options/grid.html 404
4 Change ES6 style arrow function to ES5 normal function at the following files, that may cause some low version chrome can't compatible:

userAdminCtrl.js
dashboardViewCtrl.js
widgetCtrl.js

sunil32150 committed:
Update widgetCtrl.js
